src/binder_ims_reg.c

```diff
@@ -84,14 +89,11 @@ binder_ims_reg_query_done(
     BinderBase* base = &self->base;
     BinderImsReg* ims = &self->pub;
     gboolean registered = FALSE;
-    guint32 code =
-        radio_client_aidl_interface(self->g->client) == RADIO_NETWORK_INTERFACE ?
-            RADIO_NETWORK_RESP_GET_IMS_REGISTRATION_STATE :
-            RADIO_RESP_GET_IMS_REGISTRATION_STATE;
 
     if (status != RADIO_TX_STATUS_OK) {
         ofono_error("getImsRegistrationState failed");
-    } else if (resp != code) {
+    } else if (resp != RADIO_RESP_GET_IMS_REGISTRATION_STATE &&
+        resp != RADIO_NETWORK_RESP_GET_IMS_REGISTRATION_STATE) {
```

Comment:
@monich Feel free to fix that issue or not and then merge. I tested this on both HIDL and AIDL devices and worked fine.

Reply:
Actually, this part wasn't clear to me, it's producing this compilation warning:
```
src/binder_ims_reg.c: In function 'binder_ims_reg_query_done':
src/binder_ims_reg.c:96:14: warning: comparison between 'RADIO_RESP' {aka 'enum radio_resp'} and 'enum radio_network_resp' [-Wenum-compare]
         resp != RADIO_NETWORK_RESP_GET_IMS_REGISTRATION_STATE) {
              ^~
```
and I decided to leave this warning as a reminder to check later - how come two different enums are getting mixed up. It doesn't look right from the first glance, but before I can tell how wrong it actually is, I need to check what has happened to `libgbinder-radio` API since the last time I looked at it.

Reply:
That is why the code you removed was using guint32 for the code variable, to prevent warning about comparing it to different enums. The reason for is that the RADIO_RESP enum is still used in the parameter even though it can be either RADIO_RESP or any of the AIDL interface response enum types. This a bit wrong way was to prevent duplication of the response handling functions for different interfaces.

Reply:
In any case it's not breaking anything and can be merged as is. If I have a suggestion how to do it better, we can handle it later as a separate PR.